gpu-deployment-tutorial/gpu-deployment-from-scratch.md

```diff
@@ -0,0 +1,270 @@
+
+- Install the `brev` CLI
+  - `brew install brevdev/homebrew-brev/brev && brev login`
+- Connect via SSH
+  - `brev shell <your vm name>`
+
+For Linux and Windows instructions check the [brev-cli install documentation](https://docs.nvidia.com/brev/latest/brev-cli.html#installation-instructions)
+
+### Exploring our GPU Software Environment
+
+Let's start by exploring our VM to see what software we got out of the box.
+
+```console
+$ cat /etc/os-release
+PRETTY_NAME="Ubuntu 22.04.5 LTS"
+```
+
+We can check our GPU information by running `nvidia-smi`.
+
+```console
+$ nvidia-smi
++-----------------------------------------------------------------------------------------+
+| NVIDIA-SMI 570.158.01             Driver Version: 570.158.01     CUDA Version: 12.8     |
+|-----------------------------------------+------------------------+----------------------+
+| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
+| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
+|                                         |                        |               MIG M. |
+|=========================================+========================+======================|
+|   0  NVIDIA L4                      On  |   00000000:00:03.0 Off |                    0 |
+| N/A   47C    P8             13W /   72W |       0MiB /  23034MiB |      0%      Default |
+|                                         |                        |                  N/A |
++-----------------------------------------+------------------------+----------------------+
+
++-----------------------------------------------------------------------------------------+
+| Processes:                                                                              |
+|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
+|        ID   ID                                                               Usage      |
+|=========================================================================================|
+|  No running processes found                                                             |
++-----------------------------------------------------------------------------------------+
+```
+
+Next we need the core CUDA libraries in order to run any CUDA code. Often these will be installed at the system level in `/usr/local/cuda`.
+
+```bash
+ls -ld /usr/local/cuda*
+```
+
+If these are missing we need to decide how to get those dependencies. The way we do this is different depending on whether we want to use `pip`/`uv` or `conda`/`pixi` for our Python package manager.
+
+### Python Software environments
+
+#### Pip
+
+If we want to install our packages with `pip` we need to install the CUDA core libraries at the system level, we can do this on Ubuntu with `apt`.
+
+```bash
+# Add the NVIDIA repos
+wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/cuda-keyring_1.1-1_all.deb
```

Comment:
It wasn't clear to me what this step is for, can we add some docs on what is and why `cuda-keyring_1.1-1_all.deb`